homeassistant/components/lcn/binary_sensor.py

```diff
@@ -1,21 +1,29 @@
 """Support for LCN binary sensors."""
+from __future__ import annotations
+
+from typing import Callable
+
 import pypck
 
 from homeassistant.components.binary_sensor import (
     DOMAIN as DOMAIN_BINARY_SENSOR,
     BinarySensorEntity,
 )
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_ADDRESS, CONF_DOMAIN, CONF_ENTITIES, CONF_SOURCE
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
 
 from . import LcnEntity
 from .const import BINSENSOR_PORTS, CONF_DOMAIN_DATA, SETPOINTS
-from .helpers import get_device_connection
+from .helpers import DeviceConnectionType, InputType, get_device_connection
 
 
-def create_lcn_binary_sensor_entity(hass, entity_config, config_entry):
+def create_lcn_binary_sensor_entity(
+    hass: HomeAssistantType, entity_config: ConfigType, config_entry: ConfigEntry
+) -> LcnEntity:
     """Set up an entity for this domain."""
     device_connection = get_device_connection(
-        hass, tuple(entity_config[CONF_ADDRESS]), config_entry
```

Comment:
How come the call to `tuple()` is no longer needed?

Reply:
We just need an iterable for `get_device_connection`. No need to cast to a tuple.